packages/coordinator/src/services/validation/bundle-validator.ts

```diff
@@ -158,6 +154,46 @@ export function verifyCommonBundle(
   return incorrectValueErrors
 }
 
+function verifyMedicationRequestIntents(
+  medicationRequests: Array<fhir.MedicationRequest>
+): Array<fhir.OperationOutcomeIssue> {
+  const incorrectValueErrors: Array<fhir.OperationOutcomeIssue> = []
+
+  const repeatPrescribing = isRepeatPrescribing(medicationRequests)
+  const repeatDispensing = isRepeatDispensing(medicationRequests)
+  const acute = !(repeatPrescribing || repeatDispensing)
+
+  if (acute && medicationRequests.some(
+    medicationRequest => medicationRequest.intent !== fhir.MedicationRequestIntent.ORDER
+  )) {
```

Comment:
Instead of repeating `medicationRequests.some(...)`, could you do something to make the if statements more readable? For example, something like this would avoid the repetitions:

```ts
const intentDoesNotMatch = (
  medicationRequests: fhir.MedicationRequest[],
  intent: fhir.MedicationRequestIntent
): boolean => {
  return medicationRequests.some(medicationRequest => medicationRequest.intent !== intent)
}

if (acute && intentDoesNotMatch(medicationRequests, fhir.MedicationRequestIntent.ORDER) {
  ...
}
```